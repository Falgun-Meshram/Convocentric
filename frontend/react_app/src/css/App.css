p ,a{ 
  margin: 0px; 
  color :#1C3FB5;
  font-size: 16px;
  text-align: center; 
}
a{
  text-decoration: none;
  font-size: 22px;
}
.formAlign{
  padding-left: 12rem;
  padding-right: 12rem;
  justify-content: center;
}
.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

.chat-list{
  height: 90vh;
  overflow-y: scroll !important;
}
.message-list{
  overflow-y: scroll !important;
}
.input{
  height: 10vh !important;
  overflow-y: scroll !important;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
