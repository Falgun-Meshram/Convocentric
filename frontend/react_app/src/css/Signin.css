.customButton{
    background-color: #2A4CBF!important;
    border-color: #2A4CBF !important;
    color: white !important;
    width: 100%;
  }

  .customButtonChangePassword{
    border-color: #162B71;
    background-color: #162B71;
    color: white !important;
    width: 100%;
  }

  .landing_img{
    height: 100%;
    width: 100%;
    object-fit: cover;
  }