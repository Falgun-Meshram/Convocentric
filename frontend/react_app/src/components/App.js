// import logo from '../logo.svg';
import React from 'react';
import { Col, Container, Row, Dropdown, InputGroup, DropdownButton, FormControl } from 'react-bootstrap';
import { ChatList, Input, Button, MessageList, } from 'react-chat-elements';

import '../css/App.css';
import 'react-chat-elements/dist/main.css';
import { useState } from 'react';

// const ws = new WebSocket("wss://ws.bitstamp.net");
// const apiCall = {
//   event: "bts:subscribe",
//   data: { channel: "order_book_btcusd" },
// };

// ws.onopen = (event) => {
//   console.log(`ws.onopen `);
//   console.log(event);
//   ws.send(JSON.stringify(apiCall));
// };

function App() {
  const date = new Date();
  const data = [
    {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
      id: '123'
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
      id: '456'
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    },
    {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    }, {
      avatar: 'https://facebook.github.io/react/img/logo.svg',
      alt: 'Reactjs',
      title: 'Facebook',
      subtitle: 'What are you doing?',
      date: new Date(),
      unread: 0,
    },
  ]
  const msg = [
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: date.setDate(date.getDate() - 4),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    }, {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
    {
      position: 'left',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date() - 10,
    },
    {
      position: 'right',
      type: 'text',
      text: 'Lorem ipsum dolor sit amet, consectetur adipisicing elit',
      date: new Date(),
    },
  ]
  const simpleMsg = [{
    position: 'right',
    type: 'text',
    text: 'Hey',
    date: date.setDate(date.getDate() - 4),
    replyButton: true,


  },
  {
    position: 'left',
    type: 'text',
    text: 'I am doing well',
    date: new Date(),
    replyButton: true,


  },]


  // ws.onmessage = function (event) {
  //   const json = JSON.parse(event.data);
  //   console.log(`[message] Data received from server: ${json}`);
  //   try {
  //     if ((json.event = "data")) {

  //       console.log(json.data.timestamp);
  //     }
  //   } catch (err) {
  //     //whatever you wish to do with the err
  //   }

  // };

  // const [userMessage, changeUserMessage] = useState("");
  let userMessage = ""
  const [currentGroupId, changeGroupId] = useState("");
  const handleOnChange = (e) => {
    // changeUserMessage(e.target.value)
    userMessage = e.target.value;
  };
  const handleGroupClick = (group) => {
    console.log(group);
    changeGroupId(group.id)
  };
  const handleSubmit = (e) => {
    console.log(`handlesubmit called`);
    console.log(userMessage);
  };
  const handleReply = () => {
    console.log('natural');

  }


  const getUserMessages = (groupId) => {
    console.log(`getUserMessages : ${groupId}`);
    if (groupId === "123")
      return msg;
    if (groupId === "456")
      return simpleMsg;
    else
      return [];
  };


  return (
    <Container fluid style={{ height: '100vh' }} >
      <Row style={{ height: '100vh' }}>
        <Col lg="5" style={{ height: '100vh', }}  >
          <div style={{ height: '5vh' }}>
            Profile Data
          </div>
          <ChatList
            className="chat-list"
            onClick={handleGroupClick}
            dataSource={data}
          />
        </Col>
        <Col lg="7" style={{ height: '100vh', background: 'aliceBlue' }} >
          <div style={{ height: '5vh' }}>
            Group Data
          </div>
          <div style={{ height: '85vh' }} className='message-list' >
            <MessageList
              className="message-list"
              lockable
              toBottomHeight="100%"
              dataSource={getUserMessages(currentGroupId)}
              onReplyClick={handleReply}
            />
            <Dropdown
              buttonProps={{
                text: 'Dropdown',
              }}
              items={[
                {
                  text: 'dolor'
                },
              ]} />
          </div>

          <div className='input'>
            <Input
              placeholder="Type here..."
              multiline={true}
              onChange={handleOnChange}
              rightButtons={
                <Button
                  onClick={handleSubmit}
                  color='white'
                  backgroundColor='black'
                  text='Send' />
              }
            />
          </div>
        </Col>
      </Row>
    </Container >
  );
}

export default App;
