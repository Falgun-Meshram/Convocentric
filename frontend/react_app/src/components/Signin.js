import React, { useState } from 'react';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { useNavigate } from "react-router-dom";
import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
import Tooltip from 'react-bootstrap/Tooltip'
import landing_img from '../images/landing_img.svg'
import '../css/Signin.css';

export default function Signin() {

    const navigate = useNavigate();
    const [formData, setFormData] = useState({
        username: '',
        password: ''
    })

    const redirectPage = (page) => {
        navigate(page)
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        console.log(formData);
    }

    return (
        <Row style={{ padding: '0px', margin: '0px' }} className="align-items-center">
            <Col className='background'>
            </Col>
            <Col xs={12} sm={12} md={6} lg={6} xl={6}>
                <Row style={{ padding: '0px 20px', margin: '50px 0px 0px 0px' }}>
                    <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                        <h2 style={{ textAlign: 'center', marginBottom: '3rem' }} >Convocentric</h2>
                        <h3 style={{ textAlign: 'center', marginBottom: '3rem', color: '#4566DA' }}>Welcome to Convocentric</h3>
                    </Col>
                </Row>
                <Row >
                    <Col xs={12} sm={12} md={12} lg={12} xl={12}>

                        <Form onSubmit={(e) => handleSubmit(e)}>
                            <Form.Group className="mb-3" noValidate>

                                <Form.Label style={{ color: '#4A6BDB' }}>Username</Form.Label>
                                <Form.Control required onChange={(e) => setFormData({ ...formData, username: e.target.value })} id="username" name="username" value={formData.username} type="text" placeholder="Username" />
                                <Form.Control.Feedback type="invalid">Please provide a valid name</Form.Control.Feedback>
                                {/* <Form.Text className="text-muted">
                                </Form.Text> */}
                            </Form.Group>

                            <Form.Group className="mb-3">
                                <Form.Label style={{ color: '#4A6BDB' }}>
                                    Password
                                    <OverlayTrigger
                                        placement="right"
                                        overlay={
                                            <Tooltip id="info_tooltip">
                                                Your password must be atleast 8 characters, and must contain atleast
                                                one capital letter, one small letter and one number and one special character.
                                            </Tooltip>}
                                    >
                                        <i style={{ marginLeft: '10px', color: 'green', fontSize: '20px', cursor: 'pointer' }} className="fa fa-info-circle"></i>
                                    </OverlayTrigger>
                                </Form.Label>
                                <Form.Control required pattern='(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*?&])[A-Za-z\d@$!%*?&]{8,}$'
                                    onChange={(e) => setFormData({ ...formData, password: e.target.value })} id="password" name="password" value={formData.password} type="password" placeholder="Password" />
                            </Form.Group>
                            {/* <Row style={{ margin: '0px', padding: '0px', marginTop: '-10px' }}>
                                <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                                    <p style={{ color: '#5E7DE0', cursor: 'pointer', fontSize: '14px', marginTop: '2px', textAlign: 'right' }} onClick={() => redirectPage('/forgot_password')}>
                                        Forgot Password ?
                                    </p>
                                </Col>
                            </Row> */}
                            <Row style={{ margin: '0px', padding: '0px', marginTop: '35px' }}>
                                <Col xs={12} sm={12} md={{ offset: 6, span: 6 }} lg={{ offset: 3, span: 5 }} xl={{ offset: 3, span: 5 }}>
                                    <Button size="md" className="customButton" type="submit">
                                        Sign in
                                    </Button>
                                </Col>
                            </Row>
                            <Row style={{ margin: '10px 0px 0px 0px', padding: '0px' }}>
                                <Col xs={12} sm={12} md={12} lg={12} xl={12}>
                                    <p style={{ margin: '0px', color: '#2A4CBF', padding: '15px 20px 5px 0px', fontSize: '15px', textAlign: 'center' }}>Donâ€™t have an account yet? </p>
                                    <p onClick={() => redirectPage('/signup')} style={{ cursor: 'pointer', margin: '0px', color: '#1C3FB5', padding: '0px 20px 0px 0px', fontSize: '22px', textAlign: 'center' }}>Create an account</p>
                                </Col>
                            </Row>

                        </Form>
                    </Col>
                </Row>
            </Col>
        </Row>
    )
}

