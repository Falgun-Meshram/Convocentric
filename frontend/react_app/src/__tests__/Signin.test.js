import { render, fireEvent, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import App from '../components/App';
import { rest } from 'msw'
import { setupServer } from 'msw/node'

import Signin from '../components/Signin';
import SignUp from '../components/Signup';

const server = setupServer(
    rest.post("http://localhost:4001/api/login", (req, res, ctx) => {
        return res(
            ctx.status(200),
            ctx.json({
                ok: true,
                error: ""
            }))
    })
)


describe("<Signin>", () => {
    it("Page should render", () => {
        const { container, debug } = render(
            <BrowserRouter>
                <Signin />
            </BrowserRouter>
        );
    });
    it("Input Validation: No input given", () => {
        const { container, debug } = render(
            <BrowserRouter>
                <SignUp />
            </BrowserRouter>
        );
        expect(screen.getByText("Please provide a valid name")).toBeInTheDocument();
        expect(screen.getByText("Please provide a valid email")).toBeInTheDocument();
        expect(screen.getByText("Your password must be atleast 8 characters, and must contain atleast one capital letter, one small letter and one number and one special character")).toBeInTheDocument();
        expect(screen.getByText("Password should match")).toBeInTheDocument();
    }
    )
    //TODO Add a network error test case
    it("Input Validation: Correct input given", () => {

        //TODO Add the msw to mock the api calls
        const { container, debug } = render(
            <BrowserRouter>
                <Signin />
                <App />
            </BrowserRouter>
        );
        console.log(`container ${container}`);
        const userNameInput = screen.getByPlaceholderText("Username");
        const passwordInput = screen.getByPlaceholderText("Password");

        fireEvent.change(userNameInput, { target: { value: "asd" } })
        fireEvent.change(passwordInput, { target: { value: "asdfQW!@34" } })
        const e = fireEvent.click(screen.getByText("Sign Up"))
        expect(screen.getByText("Profile Data")).toBeInTheDocument();

    }
    )
    it("Redirect to Signup", () => {
        const { container, debug } = render(
            <BrowserRouter>
                <Signin />
                <SignUp />

            </BrowserRouter>
        );
        const e = fireEvent.click(screen.getByText("Sign Up"))
        expect(screen.getByText("Already have an account?")).toBeInTheDocument()

    }
    )
});

