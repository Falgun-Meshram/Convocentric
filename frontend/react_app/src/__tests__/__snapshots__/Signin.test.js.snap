// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signin> Page should render 1`] = `
<div>
  <div
    class="align-items-center row"
    style="padding: 0px; margin: 0px;"
  >
    <div
      class="background col"
    />
    <div
      class="col-xl-6 col-lg-6 col-md-6 col-sm-12 col-12"
    >
      <div
        class="row"
        style="padding: 0px 20px; margin: 50px 0px 0px 0px;"
      >
        <div
          class="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-12"
        >
          <h2
            style="text-align: center; margin-bottom: 3rem;"
          >
            Convocentric
          </h2>
          <h3
            style="text-align: center; margin-bottom: 3rem; color: rgb(69, 102, 218);"
          >
            Welcome to Convocentric
          </h3>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-12"
        >
          <form
            class="formAlign"
          >
            <div
              class="mb-3"
              novalidate=""
            >
              <label
                class="form-label"
                style="color: rgb(74, 107, 219);"
              >
                Username
              </label>
              <input
                class="form-control"
                id="username"
                name="username"
                placeholder="Username"
                required=""
                type="text"
                value=""
              />
              <div
                class="invalid-feedback"
              >
                Please provide a valid name
              </div>
            </div>
            <div
              class="mb-3"
            >
              <label
                class="form-label"
                style="color: rgb(74, 107, 219);"
              >
                Password
                <i
                  class="fa fa-info-circle"
                  style="margin-left: 10px; color: green; font-size: 20px; cursor: pointer;"
                />
              </label>
              <input
                class="form-control"
                id="password"
                name="password"
                pattern="(?=.*[a-z])(?=.*[A-Z])(?=.*\\\\d)(?=.*[@$!%*?&])[A-Za-z\\\\d@$!%*?&]{8,}$"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              
            </div>
            <div
              class="row"
              style="margin: 35px 0px 0px 0px; padding: 0px;"
            >
              <div
                class="col-xl-5 col-lg-5 col-md-6 col-sm-12 col-12 offset-xl-3 offset-lg-3 offset-md-6"
              >
                <button
                  class="btn btn-primary btn-block btn btn-primary btn-lg"
                  type="submit"
                >
                  Sign In
                </button>
              </div>
            </div>
            
            <div
              class="row"
              style="margin: 10px 0px 0px 0px; padding: 0px;"
            >
              <div
                class="col-xl-12 col-lg-12 col-md-12 col-sm-12 col-12"
              >
                <p
                  style="margin: 0px; color: rgb(42, 76, 191); padding: 15px 20px 5px 0px; font-size: 15px; text-align: center;"
                >
                  Donâ€™t have an account yet? 
                </p>
                <p
                  style="cursor: pointer; margin: 0px; color: rgb(28, 63, 181); padding: 0px 20px 0px 0px; font-size: 22px; text-align: center;"
                >
                  Create an account
                </p>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
